Allow to conditionally compile against Qt5 by setting -DUseQt5=On

simonschmeisser opened the pull request:
This allows to conditionally compile and link moveit_setup_assistant against Qt5. It requires the respective pull requests to rviz and moveit_ros to be applied as well.

simonschmeisser committed:
Allow to conditionally compile against Qt5 by setting -DUseQt5=On

davetcoleman commented on CMakeLists.txt:
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 2.8.12)
```
Indigo's target CMake version is 2.8.11 - I'm pretty sure we shouldn't go above this or we might break Ubuntu Saucy (13.10) or something. Can you reduce the version to 11 and have it still work?
http://www.ros.org/reps/rep-0003.html#indigo-igloo-may-2014

simonschmeisser replied:
Fixed, see http://code.qt.io/cgit/qt/qtbase.git/tree/dist/changes-5.4.2 for details

I was requiring cmake 2.8.12 as rviz-qt5 and moveit-qt5 both apply against jade-devel (but the same fix should work there as well)

simonschmeisser committed:
Add line for supporting CMake 2.8.11 as required for Indigo

This appears to be only strictly necessary when using Qt >= 5.4.2 and gcc 5.x, but better safe than sorry

davetcoleman commented:
Tested on indigo-devel and 14.04 and worked fine, thanks!

davetcoleman commented:
[Build farm has warning](http://build.ros.org/job/Idoc__moveit_setup_assistant__ubuntu_trusty_amd64/changes) after this commit, but I think its unrelated:

> build succeeded, 1 warning.

which is probably from:

> /tmp/catkin_workspace/src/moveit_setup_assistant/CHANGELOG.rst:: WARNING: document isn't included in any toctree

gavanderhoorn commented on CMakeLists.txt:
```diff
@@ -23,10 +23,24 @@ include_directories(${catkin_INCLUDE_DIRS})
+else (UseQt5)
+  find_package(Qt4 REQUIRED COMPONENTS QtCore QtGui QtOpenGL)
+  include(${QT_USE_FILE})
+endif(UseQt5)
```
This form of using conditionals (`if (X) .. else (X) .. endif (X)`) is not needed, but also not recommended (anymore) in CMake (and ROS as well).

You don't restate the condition in the `else` and `endif` parts anymore.

simonschmeisser replied:
Just found the [CMake coding standards](http://docs.ros.org/jade/api/catkin/html/user_guide/standards.html) and will open another PR soonish.

Do you have a reference for the cmake part? The cmake documentation still uses this form: https://cmake.org/cmake/help/v3.5/command/if.html

>  Note that the expression in the else and endif clause is optional.

So it's optional but not deprecated nor is it recommended to omit it.
